pimd/pim_join.c

```diff
@@ -312,10 +312,6 @@ int pim_joinprune_recv(struct interface *ifp, struct pim_neighbor *neigh,
 
 			buf += addr_offset;
 
-			/* if we are filtering this group, skip the prune */
-			if (filtered)
-				continue;
-
```

Comment:
Is the reasoning here is that we should always scan the prune sources regardless of filtering rules?

Reply:
Prunes for flows for which we have no state have no effect, but if we begin filtering a (S,G) that is already being forwarded, then we'll want to continue processing prunes for that flow if it tries to prune before the flow state times out.